guides/documentations/essentials.md

```diff
@@ -0,0 +1,209 @@
+
+Metadata is used to list all components of the application, define their content, how they are displayed and specify the rules for their various properties.
+Here is how you can initialize a component:
+
+```js
+new Component({
+  id: 'id',
+  name: 'name',
+  definition: new ComponentDefinition(),
+  drawOption: new ComponentDrawOption(),
+  attributes: [new ComponentAttribute()],
+})
+```
+
+To make your plugin work, we need to have an array of [ComponentDefinitions](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/models/ComponentDefinition.js).
+Component definitions are loaded from a metadata file. Each component definition has:
+- A name
+- An icon
+- Attributes
+- ...
+
+| Left: Components list in Leto-modelizer - Middle: Related Metadata - Left: Component Panel |
+| :--: |
+| <img height="400" src="../../documentations/metadata.png"/> |
+
+Attributes of a component represent all the related properties that can/must be set inside the Component Panel (Right side of the previous image).
+Link Definitions can also be loaded from the metadata file (for colored links, dashed links...).
+
+Once the component definitions are loaded, components can be instantiated from the components list (Left side of the previous image) in Leto-modelizer.
+The DefaultMetadata class provides two methods:
+
+- `validate`: Validate the metadata.
+- `parse`: Set all component/link definitions in PluginData from your provided metadata.
+
+## Renderer
+
+The Renderer is used to serialize all the instantiated components into source files (`.tf` files for Terraform, `.yaml` files for Kubernetes, ...)
+It uses Jinja/Nunjucks templates.
+
+The DefaultRender provides these methods:
+
+- `render`: Transform all provided components in file inputs.
+- `renderFiles`: Maps all the given components into a file content and return all files.
+- `renderConfiguration`: Update configuration file content according to components data.
+
+## Parser
+
+The Parser is used to deserialize files to instantiate components and their attributes.
+It uses an external parser to work, the choice of parser depends on the plugin. For example, terrator-plugin uses [ANTLR](https://www.antlr.org/) and githubator-plugin uses [Lidy](https://github.com/ditrit/lidy-js).
+
+The DefaultParser provides these methods:
+
+- `parse`: Convert the content of files into components.
+- `getModels`: Get the list of model paths from all parsable files.
+- `parseConfiguration`: Set configuration into components.
+- `isParsable`: Indicate if this parser can parse this file.
+
+## Draw diagrams
+
+The Drawer is used to generate the diagram from components in storage.
+By default, the DefaultDrawer use [D3 library](https://d3js.org/) to draw the svg of the diagram. Only the `draw` function is called by Leto-Modelizer and used to draw all components and ComponentLinks. You can override this method if you want to use another library.
+
```

Comment:
add:
```
### Custom Layout

TODO
```

Reply:
not really pro to let a todo in a documentation like this...

Reply:
done, to check if it's ok